sdl_viewer/src/terrain_drawer/mod.rs

```diff
@@ -1,5 +1,193 @@
+use crate::c_str;
+use crate::graphic::{GlBuffer, GlProgram, GlProgramBuilder, GlUniform, GlVertexArray};
+use crate::opengl;
+use cgmath::{EuclideanSpace, Matrix4, Point3, SquareMatrix, Vector2, Zero};
+use point_viewer::math::Isometry3;
+
+use opengl::types::{GLsizeiptr, GLuint};
+
+use std::ffi::c_void;
+use std::mem;
+use std::rc::Rc;
+
 mod layer;
 mod read_write;
 
 pub use layer::TerrainLayer;
 pub use read_write::Metadata;
+
+const TERRAIN_FRAGMENT_SHADER: &str = include_str!("../../shaders/terrain.fs");
+const TERRAIN_VERTEX_SHADER: &str = include_str!("../../shaders/terrain.vs");
+const TERRAIN_GEOMETRY_SHADER: &str = include_str!("../../shaders/terrain.gs");
+
+const GRID_SIZE: u32 = 1023;
+
+#[allow(dead_code)]
```

Comment:
Yes, the buffer_position and buffer_indices fields. The reason they're stored is that they cannot be dropped or OpenGL will segfault because of their `Drop` impl. We could probably do better by storing them in the vertex array object, but I need to think about how to do that.

Reply:
Can we add this attribute only to these two fields then, instead of the whole struct?

Reply:
Already did that :)